sdk/keyvault/keyvault/test/main.c

```diff
@@ -1,18 +1,45 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
+#include <az_client_secret_credential.h>
 #include <az_keyvault.h>
 
 #include <_az_cfg.h>
 
 int exit_code = 0;
 
 int main() {
-  // Creates keyvault client and init it
+  // Creates keyvault client
   az_keyvault_keys_client client;
-  az_keyvault_keys_client_options options;
-  az_span uri = { 0 };
 
-  az_result operation_result = az_keyvault_keys_client_init(&client, uri, &options);
+  // create credentials as client_id type
+  az_client_secret_credential auth = { 0 };
+  // Get secrets from ENV
+  char * const TENANT_ID = getenv("tenant_id");
+  char * const CLIENT_ID = getenv("client_id");
+  char * const CLIENT_SECRET = getenv("client_secret");
+  // init auth_credentials struc
+  az_result const creds_retcode = az_client_secret_credential_init(
+      &auth,
+      (az_span){ .begin = TENANT_ID, .size = strlen(TENANT_ID) },
+      (az_span){ .begin = CLIENT_ID, .size = strlen(CLIENT_ID) },
+      (az_span){ .begin = CLIENT_SECRET, .size = strlen(CLIENT_SECRET) });
+
+  // Create client options
+  az_keyvault_keys_client_options options = { .version = AZ_STR("7.0") };
+
+  // Init client
+  // TODO: introduce init and init_with_options so options can be optional for user
+  az_result operation_result = az_keyvault_keys_client_init(
+      &client, AZ_STR("https://testingforc99.vault.azure.net"), (az_credential *)&auth, &options);
```

Comment:
1. Can you also gert key vault URL from environment?
2. Do you really have to do this (az_credential *) cast?

Reply:
(az_credential *) cast is for Windows, if I remember well